### ISSUE TITLE
Improve description of LocalDate

### ISSUE THREAD

**Issue by cmeeren** [ChilliCream/graphql-platform#7636]
### Product

Hot Chocolate

### Is your feature request related to a problem?

The LocalDate scalar (presumably from HotChocolate.Types.NodaTime) has this description in the GraphQL schema:

> LocalDate is an immutable struct representing a date within the calendar, with no reference to a particular time zone or time of day.

It doesn't make sense to refer to things like "immutable" or "struct" for GraphQL types.

### The solution you'd like

Change the description so it makes sense from a GraphQL perspective. For example:

> LocalDate is a String-based scalar with the format `YYYY-MM-DD` that represents a date within the calendar, with no reference to a particular time zone or time of day.

Note that this may apply to other NodaTime scalars, too (I haven't checked).

**Comment by michaelstaib** [comment]
I disagree with this change as it talks about the json serialization which is not its runtime representation ... GraphQL is not fixed to that and you could also have a binary representation of the struct or even the struct as a representation. So `is a String-based scalar with` is  factual wrong.

**Comment by cmeeren** [comment]
Oh, come on... You don't have to agree with specifics, but at least work with me instead of just shooting me down and not even addressing the main point 🙃

Firstly, according to my understanding, saying it is String-based is not related to JSON. String is a built-in GraphQL type, which I assume is true regardless of the representation.

Secondly, I did not come up with this. See the official templates for String-based community scalars: https://scalars.graphql.org And for a concrete example, see e.g. the DateTime scalar: https://scalars.graphql.org/andimarek/date-time Both places use the concept of being String-based.

Thirdly, regardless of the above, I assume from a pragmatic standpoint that most GraphQL APIs using HotChocolate will use JSON. Not documenting the format of custom scalars is actively unhelpful. It requires API client devs to guess the format, or to call the API to see example values and make assumptions about the format. Again, see the community scalars, where the format is documented. 

Fourthly, regardless of even that: As I wrote in the OP, it was just an example. The main point, as I described, is about not copying the NodaTime doc string verbatim, with its references to immutability and being a struct, which definitely doesn't make sense in GraphQL.
